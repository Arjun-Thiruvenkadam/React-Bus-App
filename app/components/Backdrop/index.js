/**
 *
 * Backdrop
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import './backdrop.scss';

const Backdrop = props => {
  return props.show ? (
    <div className={'backdrop'} onClick={props.clicked} />
  ) : null;
};

Backdrop.propTypes = {
  show: PropTypes.bool.isRequired,
  clicked: PropTypes.func.isRequired,
};

export default Backdrop;
