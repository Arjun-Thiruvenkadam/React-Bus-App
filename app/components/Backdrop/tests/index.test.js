import React from 'react';
import { render } from 'react-testing-library';
import Backdrop from '../index';

describe('<Backdrop />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Backdrop show clicked={null} />);
    expect(firstChild).toMatchSnapshot();
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Backdrop show={false} clicked={null} />);
    expect(firstChild).toMatchSnapshot();
  });
});
