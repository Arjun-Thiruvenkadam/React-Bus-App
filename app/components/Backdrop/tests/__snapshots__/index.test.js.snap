// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Backdrop /> should render and match the snapshot 1`] = `
<div
  class="backdrop"
/>
`;

exports[`<Backdrop /> should render and match the snapshot 2`] = `null`;
